### DISCUSSION TITLE
Quantization and Dequantization

### DISCUSSION THREAD

**Discussion by Mad-Roy** [microsoft/onnxruntime#22024] [Other Q&A]
In  the Federated Learning settings I want to use the quantization techniques to improve communication efficiency. I was able to quantize a YOLO model to UInt8. But I did'n find how to dequantize the YOLO model back to float32. Is it possible with onnx and when yes, then how? Thanks in advance

**Comment by xadupre** [comment]
This was not implemented, mostly because it is impossible to restore the original weights once they were quantized. Any particular reason you want to do this?

**Comment by Mad-Roy** [comment]
I want to improve the communication, because a quantized model is smaller for transmitting, but later or in next learning rounds (in Federated Learning) I want to continue learning. In this paper (https://arxiv.org/abs/2312.15186), the quantization was used as a compression technique, I think, it is a good idea
